### PR TITLE
feat: [PAGOPA-1455] migration to java 17

### PR DESCRIPTION
**Pull request by alessio-acitelli**
#### List of Changes

- update to java 17

#### Motivation and Context

- sonar analysis has deprecated java 11

#### How Has This Been Tested?

- [Deploy](https://github.com/pagopa/pagopa-debt-position/actions/runs/7558469100) and run [integration test](https://github.com/pagopa/pagopa-debt-position/actions/runs/7558572929) in DEV env

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: Dockerfile.deploy [hunk 1]
```diff
@@ -1,10 +1,10 @@
-FROM adoptopenjdk/openjdk11:alpine-jre as builder
+FROM eclipse-temurin:17-jdk-alpine as builder
```
Comment (AngeloCaporaso) [inline Dockerfile.deploy:1]: Is it required to have this version or is it better to align everything to `amazoncorretto`?

File: Dockerfile.test [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM --platform=linux/amd64 adoptopenjdk/openjdk11:jdk-11.0.9.1_1-alpine
+FROM --platform=linux/amd64 eclipse-temurin:17-jdk-alpine
```
Comment (AngeloCaporaso) [inline Dockerfile.test:1]: as above

**Review by AngeloCaporaso — APPROVED** [review]
Just a few questions, anyway, LGTM 🔝

**Review by FedericoRuzzier — APPROVED** [review]
LGTM 👍

**Pull request merged**